apps/picsa-tools/climate-tool/src/app/services/climate-data.service.ts

```diff
@@ -28,6 +28,41 @@ export class ClimateDataService {
 
   constructor(private configurationService: ConfigurationService) {
     this.stationDataCache = arrayToHashmap(DATA.HARDCODED_STATIONS, 'id');
+    this.loadPreferredStation();
+  }
+
+  /** Retrieve the preferred station ID from user settings */
+  public getPreferredStationId(): string | undefined {
+    const userSettings = this.configurationService.userSettings();
+    return userSettings.climate_tool?.station_id;
+  }
+
+  /** Load user preferred station from settings */
+  private loadPreferredStation() {
+    const userSettings = this.configurationService.userSettings();
+    if (userSettings.climate_tool?.station_id) {
+      const stationId = userSettings.climate_tool.station_id;
+      const stationExists = this.stationHashmap()[stationId];
+      if (!stationExists) {
+        console.error(`Saved station ID ${stationId} is invalid.`);
+      }
+    }
+  }
+
+  /** Allow user to set preferred station */
+  public setPreferredStation(stationID: string) {
+    const stationExists = this.stationHashmap()[stationID];
+    if (!stationExists) {
+      console.error(`Station with ID ${stationID} does not exist.`);
```

Comment:
nit(blocking): this is the bug I mentioned before - we don't actually want to check if the station exists as we want a user to just be able to remove their station by setting as something either `undefined` or `null` (or just an empty string). Once users can remove their preferred station we don't need the extra `forceSelect` param